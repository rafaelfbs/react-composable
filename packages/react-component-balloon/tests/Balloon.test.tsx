import * as React from "react";
import * as renderer from 'react-test-renderer';
import Balloon from "../src";

describe('<Balloon />', () => {
    describe('When received colors', () => {
        it('renders correctly', () => {
            const tree = renderer
                .create(
                    <Balloon
                        message="My balloon message"
                        balloonColor="red"
                        balloonBackgroundColor="gray"
                        balloonBorderColor="gainsboro"
                    />
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('When balloon is positioned on top', () => {
        it('renders correctly with default props', () => {
            const tree = renderer
                .create(
                    <Balloon
                        message="My balloon message"
                        balloonArrowPosition="top"
                    />
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('renders correctly with offset at start', () => {
            const tree = renderer
                .create(
                    <Balloon
                        message="My balloon message"
                        balloonArrowPosition="top"
                        balloonArrowPositionOffset="0"
                    />
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('renders correctly with offset at end', () => {
            const tree = renderer
                .create(
                    <Balloon
                        message="My balloon message"
                        balloonArrowPosition="top"
                        balloonArrowPositionOffset="100%"
                    />
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('When balloon is positioned on right', () => {
        it('renders correctly with default props', () => {
            const tree = renderer
                .create(
                    <Balloon
                        message="My balloon message"
                        balloonArrowPosition="right"
                    />
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('renders correctly with offset at start', () => {
            const tree = renderer
                .create(
                    <Balloon
                        message="My balloon message"
                        balloonArrowPosition="right"
                        balloonArrowPositionOffset="0"
                    />
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('renders correctly with offset at end', () => {
            const tree = renderer
                .create(
                    <Balloon
                        message="My balloon message"
                        balloonArrowPosition="right"
                        balloonArrowPositionOffset="100%"
                    />
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('When balloon is positioned on bottom', () => {
        it('renders correctly with default props', () => {
            const tree = renderer
                .create(
                    <Balloon
                        message="My balloon message"
                        balloonArrowPosition="bottom"
                    />
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('renders correctly with offset at start', () => {
            const tree = renderer
                .create(
                    <Balloon
                        message="My balloon message"
                        balloonArrowPosition="bottom"
                        balloonArrowPositionOffset="0"
                    />
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('renders correctly with offset at end', () => {
            const tree = renderer
                .create(
                    <Balloon
                        message="My balloon message"
                        balloonArrowPosition="bottom"
                        balloonArrowPositionOffset="100%"
                    />
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });

    describe('When balloon is positioned on left', () => {
        it('renders correctly with default props', () => {
            const tree = renderer
                .create(
                    <Balloon
                        message="My balloon message"
                        balloonArrowPosition="left"
                    />
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('renders correctly with offset at start', () => {
            const tree = renderer
                .create(
                    <Balloon
                        message="My balloon message"
                        balloonArrowPosition="left"
                        balloonArrowPositionOffset="0"
                    />
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });

        it('renders correctly with offset at end', () => {
            const tree = renderer
                .create(
                    <Balloon
                        message="My balloon message"
                        balloonArrowPosition="right"
                        balloonArrowPositionOffset="100%"
                    />
                )
                .toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
});
