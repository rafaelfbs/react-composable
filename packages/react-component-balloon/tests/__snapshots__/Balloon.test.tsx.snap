// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Balloon /> When balloon is positioned on bottom renders correctly with default props 1`] = `
<div
  className="rc-balloon"
  style={
    Object {
      "backgroundColor": "black",
      "border": "1px solid black",
      "color": "white",
      "padding": "5px",
      "position": "relative",
    }
  }
>
  <span
    className="rc-balloon-arrow"
    style={
      Object {
        "bottom": "100%",
        "height": "10px",
        "left": "calc(50% - 10px)",
        "position": "absolute",
        "width": "20px",
        "zIndex": 10,
      }
    }
  >
    <svg
      height={10}
      style={
        Object {
          "verticalAlign": "top",
        }
      }
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="black"
        points="0,10 10,0 20,10"
      />
      <polyline
        fill="transparent"
        points="0,10 10,0 20,10"
        stroke="black"
      />
    </svg>
  </span>
  My balloon message
</div>
`;

exports[`<Balloon /> When balloon is positioned on bottom renders correctly with offset at end 1`] = `
<div
  className="rc-balloon"
  style={
    Object {
      "backgroundColor": "black",
      "border": "1px solid black",
      "color": "white",
      "padding": "5px",
      "position": "relative",
    }
  }
>
  <span
    className="rc-balloon-arrow"
    style={
      Object {
        "bottom": "100%",
        "height": "10px",
        "left": "calc(100% - 10px)",
        "position": "absolute",
        "width": "20px",
        "zIndex": 10,
      }
    }
  >
    <svg
      height={10}
      style={
        Object {
          "verticalAlign": "top",
        }
      }
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="black"
        points="0,10 10,0 20,10"
      />
      <polyline
        fill="transparent"
        points="0,10 10,0 20,10"
        stroke="black"
      />
    </svg>
  </span>
  My balloon message
</div>
`;

exports[`<Balloon /> When balloon is positioned on bottom renders correctly with offset at start 1`] = `
<div
  className="rc-balloon"
  style={
    Object {
      "backgroundColor": "black",
      "border": "1px solid black",
      "color": "white",
      "padding": "5px",
      "position": "relative",
    }
  }
>
  <span
    className="rc-balloon-arrow"
    style={
      Object {
        "bottom": "100%",
        "height": "10px",
        "left": "calc(0 - 10px)",
        "position": "absolute",
        "width": "20px",
        "zIndex": 10,
      }
    }
  >
    <svg
      height={10}
      style={
        Object {
          "verticalAlign": "top",
        }
      }
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="black"
        points="0,10 10,0 20,10"
      />
      <polyline
        fill="transparent"
        points="0,10 10,0 20,10"
        stroke="black"
      />
    </svg>
  </span>
  My balloon message
</div>
`;

exports[`<Balloon /> When balloon is positioned on left renders correctly with default props 1`] = `
<div
  className="rc-balloon"
  style={
    Object {
      "backgroundColor": "black",
      "border": "1px solid black",
      "color": "white",
      "padding": "5px",
      "position": "relative",
    }
  }
>
  <span
    className="rc-balloon-arrow"
    style={
      Object {
        "height": "20px",
        "left": "100%",
        "position": "absolute",
        "top": "calc(50% - 10px)",
        "width": "10px",
        "zIndex": 10,
      }
    }
  >
    <svg
      height={20}
      style={
        Object {
          "verticalAlign": "top",
        }
      }
      width={10}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="black"
        points="0,20 10,10 0,0"
      />
      <polyline
        fill="transparent"
        points="0,20 10,10 0,0"
        stroke="black"
      />
    </svg>
  </span>
  My balloon message
</div>
`;

exports[`<Balloon /> When balloon is positioned on left renders correctly with offset at end 1`] = `
<div
  className="rc-balloon"
  style={
    Object {
      "backgroundColor": "black",
      "border": "1px solid black",
      "color": "white",
      "padding": "5px",
      "position": "relative",
    }
  }
>
  <span
    className="rc-balloon-arrow"
    style={
      Object {
        "height": "20px",
        "position": "absolute",
        "right": "100%",
        "top": "calc(100% - 10px)",
        "width": "10px",
        "zIndex": 10,
      }
    }
  >
    <svg
      height={20}
      style={
        Object {
          "verticalAlign": "top",
        }
      }
      width={10}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="black"
        points="10,20 0,10 10,0"
      />
      <polyline
        fill="transparent"
        points="10,20 0,10 10,0"
        stroke="black"
      />
    </svg>
  </span>
  My balloon message
</div>
`;

exports[`<Balloon /> When balloon is positioned on left renders correctly with offset at start 1`] = `
<div
  className="rc-balloon"
  style={
    Object {
      "backgroundColor": "black",
      "border": "1px solid black",
      "color": "white",
      "padding": "5px",
      "position": "relative",
    }
  }
>
  <span
    className="rc-balloon-arrow"
    style={
      Object {
        "height": "20px",
        "left": "100%",
        "position": "absolute",
        "top": "calc(0 - 10px)",
        "width": "10px",
        "zIndex": 10,
      }
    }
  >
    <svg
      height={20}
      style={
        Object {
          "verticalAlign": "top",
        }
      }
      width={10}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="black"
        points="0,20 10,10 0,0"
      />
      <polyline
        fill="transparent"
        points="0,20 10,10 0,0"
        stroke="black"
      />
    </svg>
  </span>
  My balloon message
</div>
`;

exports[`<Balloon /> When balloon is positioned on right renders correctly with default props 1`] = `
<div
  className="rc-balloon"
  style={
    Object {
      "backgroundColor": "black",
      "border": "1px solid black",
      "color": "white",
      "padding": "5px",
      "position": "relative",
    }
  }
>
  <span
    className="rc-balloon-arrow"
    style={
      Object {
        "height": "20px",
        "position": "absolute",
        "right": "100%",
        "top": "calc(50% - 10px)",
        "width": "10px",
        "zIndex": 10,
      }
    }
  >
    <svg
      height={20}
      style={
        Object {
          "verticalAlign": "top",
        }
      }
      width={10}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="black"
        points="10,20 0,10 10,0"
      />
      <polyline
        fill="transparent"
        points="10,20 0,10 10,0"
        stroke="black"
      />
    </svg>
  </span>
  My balloon message
</div>
`;

exports[`<Balloon /> When balloon is positioned on right renders correctly with offset at end 1`] = `
<div
  className="rc-balloon"
  style={
    Object {
      "backgroundColor": "black",
      "border": "1px solid black",
      "color": "white",
      "padding": "5px",
      "position": "relative",
    }
  }
>
  <span
    className="rc-balloon-arrow"
    style={
      Object {
        "height": "20px",
        "position": "absolute",
        "right": "100%",
        "top": "calc(100% - 10px)",
        "width": "10px",
        "zIndex": 10,
      }
    }
  >
    <svg
      height={20}
      style={
        Object {
          "verticalAlign": "top",
        }
      }
      width={10}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="black"
        points="10,20 0,10 10,0"
      />
      <polyline
        fill="transparent"
        points="10,20 0,10 10,0"
        stroke="black"
      />
    </svg>
  </span>
  My balloon message
</div>
`;

exports[`<Balloon /> When balloon is positioned on right renders correctly with offset at start 1`] = `
<div
  className="rc-balloon"
  style={
    Object {
      "backgroundColor": "black",
      "border": "1px solid black",
      "color": "white",
      "padding": "5px",
      "position": "relative",
    }
  }
>
  <span
    className="rc-balloon-arrow"
    style={
      Object {
        "height": "20px",
        "position": "absolute",
        "right": "100%",
        "top": "calc(0 - 10px)",
        "width": "10px",
        "zIndex": 10,
      }
    }
  >
    <svg
      height={20}
      style={
        Object {
          "verticalAlign": "top",
        }
      }
      width={10}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="black"
        points="10,20 0,10 10,0"
      />
      <polyline
        fill="transparent"
        points="10,20 0,10 10,0"
        stroke="black"
      />
    </svg>
  </span>
  My balloon message
</div>
`;

exports[`<Balloon /> When balloon is positioned on top renders correctly with default props 1`] = `
<div
  className="rc-balloon"
  style={
    Object {
      "backgroundColor": "black",
      "border": "1px solid black",
      "color": "white",
      "padding": "5px",
      "position": "relative",
    }
  }
>
  <span
    className="rc-balloon-arrow"
    style={
      Object {
        "height": "10px",
        "left": "calc(50% - 10px)",
        "position": "absolute",
        "top": "100%",
        "width": "20px",
        "zIndex": 10,
      }
    }
  >
    <svg
      height={10}
      style={
        Object {
          "verticalAlign": "top",
        }
      }
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="black"
        points="0,0 10,10 20,0"
      />
      <polyline
        fill="transparent"
        points="0,0 10,10 20,0"
        stroke="black"
      />
    </svg>
  </span>
  My balloon message
</div>
`;

exports[`<Balloon /> When balloon is positioned on top renders correctly with offset at end 1`] = `
<div
  className="rc-balloon"
  style={
    Object {
      "backgroundColor": "black",
      "border": "1px solid black",
      "color": "white",
      "padding": "5px",
      "position": "relative",
    }
  }
>
  <span
    className="rc-balloon-arrow"
    style={
      Object {
        "height": "10px",
        "left": "calc(100% - 10px)",
        "position": "absolute",
        "top": "100%",
        "width": "20px",
        "zIndex": 10,
      }
    }
  >
    <svg
      height={10}
      style={
        Object {
          "verticalAlign": "top",
        }
      }
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="black"
        points="0,0 10,10 20,0"
      />
      <polyline
        fill="transparent"
        points="0,0 10,10 20,0"
        stroke="black"
      />
    </svg>
  </span>
  My balloon message
</div>
`;

exports[`<Balloon /> When balloon is positioned on top renders correctly with offset at start 1`] = `
<div
  className="rc-balloon"
  style={
    Object {
      "backgroundColor": "black",
      "border": "1px solid black",
      "color": "white",
      "padding": "5px",
      "position": "relative",
    }
  }
>
  <span
    className="rc-balloon-arrow"
    style={
      Object {
        "height": "10px",
        "left": "calc(0 - 10px)",
        "position": "absolute",
        "top": "100%",
        "width": "20px",
        "zIndex": 10,
      }
    }
  >
    <svg
      height={10}
      style={
        Object {
          "verticalAlign": "top",
        }
      }
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="black"
        points="0,0 10,10 20,0"
      />
      <polyline
        fill="transparent"
        points="0,0 10,10 20,0"
        stroke="black"
      />
    </svg>
  </span>
  My balloon message
</div>
`;

exports[`<Balloon /> When received a background color renders correctly 1`] = `
<div
  className="rc-balloon"
  style={
    Object {
      "backgroundColor": "gray",
      "border": "1px solid black",
      "color": "white",
      "padding": "5px",
      "position": "relative",
    }
  }
>
  <span
    className="rc-balloon-arrow"
    style={
      Object {
        "height": "10px",
        "left": "calc(50% - 10px)",
        "position": "absolute",
        "top": "100%",
        "width": "20px",
        "zIndex": 10,
      }
    }
  >
    <svg
      height={10}
      style={
        Object {
          "verticalAlign": "top",
        }
      }
      width={20}
      xmlns="http://www.w3.org/2000/svg"
    >
      <polygon
        fill="gray"
        points="0,0 10,10 20,0"
      />
      <polyline
        fill="transparent"
        points="0,0 10,10 20,0"
        stroke="black"
      />
    </svg>
  </span>
  My balloon message
</div>
`;
